### PR TITLE
Move auth_secret and recovery codes to TFA embed field

### PR DESCRIPTION
**Pull request by starbelly**
This PR is a follow up to #866. 

 - Refactor RecoveryCode to embedded schema
 - Refactor Recovery to work with lists of RecoveryCode and binaries.
 - Added TFA embedded schema
 - Added Users.tfa_recover/2
 - Added User.set_recovery_code_used_at/1
 - Added tests for update_security/2 in user_test.exs
 - Changed maybe_update_auth_secret/2 to maybe_update_tfa/2
 - Changed update_security/2 to create or destroy a tfa schema
 - Removed migration to add auth_secret on user
 - Removed migration to create user_recovery_codes

### CODE REVIEW

**Comment by ericmj** [comment]
@starbelly @supersimple Should this be reviewed or is it blocking on anything else?

**Comment by starbelly** [comment]
@ericmj Nothing blocking this except a review and merge into @supersimple's branch.

File: lib/hexpm/accounts/users.ex [hunk 1]
```diff
@@ -383,6 +385,18 @@ defmodule Hexpm.Accounts.Users do
     end
   end
 
+  def tfa_recover(%User{} = user, code_str) do
+    case Recovery.verify(user.tfa.recovery_codes, code_str) do
+      {:ok, %RecoveryCode{} = code} ->
```
Comment (achempion) [inline lib/hexpm/accounts/users.ex]: will it match ever if `Repo.update!/2` doesn't return struct but schema directly or raises an error

Comment (achempion) [reply lib/hexpm/accounts/users.ex]: cc @supersimple @starbelly

**Pull request merged**